### DISCUSSION TITLE
Finding partner shared and saved images

### DISCUSSION THREAD

**Discussion by leocrawford** [xob0t/Google-Photos-Toolkit#13] [General]
Hi. I think like many people (judging by the online discussions) shared my google photos album with my partner, set their device to autosave and instantly regretted it. What I want to do is delete every media item that has been saved - but it is tricky to identify them.

I'd previously used the API to find all dupes between my photos and theirs (based on Name and ModTime) and thus generated a set of IDs to delete.. but turns out API won't allow you to delete them (or even move them to another album where they can be easily deleted).

Looking at the website I can see that saved photos are initially marked as "Shared by <name>" and then have "Saved to your photos" added when saved, so they should be possible to identify.

Now looking at the website protocol (if you can call it that.. I'm nightly impressed you have managed to make sense of it) I can see my name in the JSON (along with my gender!), and before being saved I can see "rclone" (the app I originally uploaded images with) though this seems to disapear once saved.

My question is, is there a way of parsing out that "shared by" and "saved" status in order to perform actions on those? And if not is there a way I can provide a (very long) list of IDs to have those deleted? Any ideas/feedback really welcome..

Thanks hugely for the code.. it's the most hopeful path I've seen in sometime.

**Comment by xob0t** [comment]
Hi. This looks similar to #8
GPTK does not support it atm, but it is theoretically possible

**Comment by leocrawford** [comment]
Thanks. Given that is there a way of just providing a bunch of IDs?

And as an aside, is it possible to see the results of a query without perform an action (e.g. dry run)?

**Reply by xob0t** [reply]
> Thanks. Given that is there a way of just providing a bunch of IDs?

No, but I've just pushed an update so gptk's api class can be accessed globally
https://github.com/xob0t/Google-Photos-Toolkit?tab=readme-ov-file#use-gptks-api

> And as an aside, is it possible to see the results of a query without perform an action (e.g. dry run)?

Also no, but "add to album" can be used as a substitute in most cases.

**Reply by jindalshreyansh** [reply]
Would this work, instead of trash:
gptkApi.addItemsToAlbum([itemInfo.dedupKey], albumName = "NotOwned");

Also, in info it says
Shared by a Google user
Saved to your photos
Who would the owner be?

**Comment by Sherbeeny** [comment]
I have used Google Jules to build this:
[Google Photos Unsaved Finder](https://github.com/Sherbeeny/Google-Photos-Unsaved-Finder)

Hope it helps and looking forward to your feedback.
